go/runtime/host/tdx/qemu.go

```diff
@@ -170,18 +177,24 @@ func (p *qemuProvisioner) getSandboxConfig(cfg host.Config, _ sandbox.Connector,
 			"-smp", fmt.Sprintf("%d", resources.CPUCount),
 			"-name", fmt.Sprintf("oasis-%s-%s", cfg.ID, cfg.Component.ID()),
 			"-cpu", "host",
-			"-machine", "q35,kernel-irqchip=split,confidential-guest-support=tdx,hpet=off",
-			"-bios", firmware,
 			"-nographic",
 			"-nodefaults",
 			// Serial port.
 			"-serial", "stdio",
 			"-device", "virtio-serial,max_ports=1",
+		},
+	}
+
+	if !p.cfg.InsecureMock {
+		// TDX mode.
+		pcfg.Args = append(pcfg.Args,
+			"-machine", "q35,kernel-irqchip=split,confidential-guest-support=tdx,hpet=off",
+			"-bios", firmware,
 			// TDX remote attestation via VSOCK.
 			"-object", `{"qom-type":"tdx-guest","id":"tdx","quote-generation-socket":{"type":"vsock","cid":"2","port":"4050"}}`,
 			// VSOCK.
 			"-device", fmt.Sprintf("vhost-vsock-pci,guest-cid=%d", cid),
```

Comment:
You need VSOCK even for the non-TDX version. You just don't need the above remote attestation socket.